pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -597,6 +597,13 @@ func (nc *NodeController) deletePods(nodeID string) error {
 		if pod.Status.Host != nodeID {
 			continue
 		}
+		if api.RestartPolicyNever == pod.Spec.RestartPolicy {
+			continue
+		} else if api.RestartPolicyOnFailure == pod.Spec.RestartPolicy {
+			if api.PodSucceeded == pod.Status.Phase {
```

Comment:
Thanks for your reply.
Please see this use case:
  We create a pod called myPod which is used to process a text file and its RestartPolicy is OnFailure. If the pod's status is PodSucceeded, we think the processing is complete. 
  Now we assume that a node has been crashed, the NodeController may delete all the pods (include myPod) on this node. When the node reboot, myPod will be run twice.